### 1
В предыдущих главах мы работали с данными без их предварительной очистки.

В этой и следующих задачах мы используем набор данных с фильмами и сериалами компании Disney. CSV файл доступен под названием 'disney_title.csv'. Ознакомиться более подробно с датасетом можно по следующей ссылке: https://www.kaggle.com/datasets/shivamb/disney-movies-and-tv-shows.

В этой задаче выведите информацию о датасете, используя функцию .info()

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('disney_title.csv')
data.head(5)

type                                             title  \
0    Movie  Duck the Halls: A Mickey Mouse Christmas Special   
1    Movie                            Ernest Saves Christmas   
2    Movie                      Ice Age: A Mammoth Christmas   
3    Movie                        The Queen Family Singalong   
4  TV Show                             The Beatles: Get Back   

                            director  \
0  Alonso Ramirez Ramos, Dave Wasson   
1                        John Cherry   
2                       Karen Disher   
3                    Hamish Hamilton   
4                                NaN   

                                                cast        country  \
0  Chris Diamantopoulos, Tony Anselmo, Tress MacN...            NaN   
1           Jim Varney, Noelle Parker, Douglas Seale            NaN   
2  Raymond Albert Romano, John Leguizamo, Denis L...  United States   
3  Darren Criss, Adam Lambert, Derek Hough, Alexa...            NaN   
4  John Lennon, Paul McCartney, George Harrison, ...            NaN   

   release_year rating  duration                      listed_in  \
0          2016   TV-G    23 min              Animation, Family   
1          1988     PG    91 min                         Comedy   
2          2011   TV-G    23 min      Animation, Comedy, Family   
3          2021  TV-PG    41 min                        Musical   
4          2021    NaN  1 Season  Docuseries, Historical, Music   

                                         description        Date  
0   Join Mickey and the gang as they duck the halls!  2021-11-26  
1  Santa Claus passes his magic bag to a new St. ...  2021-11-26  
2          Sid the Sloth is on Santa's naughty list.  2021-11-26  
3               This is real life, not just fantasy!  2021-11-26  
4  A three-part documentary from Peter Jackson ca...  2021-11-25

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1450 entries, 0 to 1449
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   type          1450 non-null   object
 1   title         1450 non-null   object
 2   director      977 non-null    object
 3   cast          1260 non-null   object
 4   country       1231 non-null   object
 5   release_year  1450 non-null   int64 
 6   rating        1447 non-null   object
 7   duration      1450 non-null   object
 8   listed_in     1450 non-null   object
 9   description   1450 non-null   object
 10  Date          1447 non-null   object
dtypes: int64(1), object(10)
memory usage: 124.7+ KB


### 2
В предыдущем задании мы обнаружили, что переменная Date из датасета по фильмам и сериалам компании Disney имеет тип object. В этом задании мы поменяем тип переменной с object на datetime с помощью функции pd.to_datetime(). В выводе распечатайте тип столбца ['Date'] с помощью .dtype

In [4]:
data['Date'] = pd.to_datetime(data['Date'])
print(data['Date'].dtype)

datetime64[ns]


### 3
Отфильтруйте все фильмы и сериалы, которые выходили с 1 января 2020 года по 1 января 2021 года. В результате выведите переменную с названиями десяти первых фильмов после фильтрации, с помощью функции head().

In [5]:
films_2020 = data[(data['Date'] >= '2020-01-01')&(data['Date'] <= '2021-01-01')].sort_values(by='Date')

In [6]:
films_2020['title'].head(10)

652                      Cars Toon: Time Travel Mater
646                                     Austin & Ally
647    Billy Dilley's Super-Duper Subterranean Summer
648                            Cars Toon: El Materdor
649                             Cars Toon: Moon Mater
650                     Cars Toon: Rescue Squad Mater
651                               Cars Toon: Spinning
653                            Cars Toon: Tokyo Mater
654              Cars Toon: Unidentified Flying Mater
655                                Cars Toons: Bugged
Name: title, dtype: object

### 4
Вам необходимо удалить столбец из датасета 'release_year' и вывести название оставшихся столбцов с помощью функции list().

In [7]:
data.drop('release_year', axis=1, inplace=True)

In [8]:
print(list(data.columns))

['type', 'title', 'director', 'cast', 'country', 'rating', 'duration', 'listed_in', 'description', 'Date']


### 5
 В этом задании вам необходимо поменять название переменных из 'disney_title.csv' на такие же названия написанные с большой буквы. В результате выведите названия стобцов с помощью функции list()

In [9]:
new_columns = []

In [10]:
for column in data.columns:
    column = column.capitalize()
    new_columns.append(column)

In [11]:
data.columns = new_columns

In [12]:
print(list(data.columns))

['Type', 'Title', 'Director', 'Cast', 'Country', 'Rating', 'Duration', 'Listed_in', 'Description', 'Date']


### 6
В этом задании вам необходимо заменить "&" на "," в столбце "listed_in", сохранить результат в новую переменную вывести последнии пять значений в новой переменной с помощью функции 'tail()'.

In [13]:
data['Listed_in'] = data['Listed_in'].str.replace('&', ',')
print(data['Listed_in'].tail())

1445        Action-Adventure, Family, Science Fiction
1446                 Action-Adventure, Comedy, Family
1447                      Biographical, Comedy, Drama
1448                     Buddy, Comedy, Coming of Age
1449    Action-Adventure, Animals , Nature, Animation
Name: Listed_in, dtype: object


### 7
В этой задаче требуется вывести количество пропущенных значений по каждому стобцу.

In [14]:
print(data.isna().sum())

Type             0
Title            0
Director       473
Cast           190
Country        219
Rating           3
Duration         0
Listed_in        0
Description      0
Date             3
dtype: int64


### 8
По результатам седьмого задания мы выяснили, что небольшое количество пропущенных значений (по три) находятся в столбцах 'rating' и 'Date'. Воспользуйтесь функцией 'dropna()' для удаление пропущенных значений во всех столбцах. В результате выведите количество пропущенных переменных по каждому столбцу в вашем датасете.

In [15]:
drop_data = data.dropna()
print(drop_data.isna().sum())

Type           0
Title          0
Director       0
Cast           0
Country        0
Rating         0
Duration       0
Listed_in      0
Description    0
Date           0
dtype: int64


### 9
Выведите процент пропущенных значений относительного всех значений. Результат округлите до сотых с помощью функции "round()".

In [16]:
print(round(data.isna().mean()*100, 2))

Type            0.00
Title           0.00
Director       32.62
Cast           13.10
Country        15.10
Rating          0.21
Duration        0.00
Listed_in       0.00
Description     0.00
Date            0.21
dtype: float64


### 10
Из предыдущего задания мы выяснили, что наибольшие проценты пропущенных значений в CSV файле 'disney_title.csv' находятся в стобцах director (32.62%), cast (13.10%) и country (15.10%). В этом задании заменить пропущенные значения в стобце 'country' на 'Country not specified' с помощью функции 'fillna'. Выведите первые пять первых значений переменной 'country'.

In [17]:
data['Country'] = data['Country'].fillna('Country not specified')
data['Country'].head()

0    Country not specified
1    Country not specified
2            United States
3    Country not specified
4    Country not specified
Name: Country, dtype: object